homeassistant/components/velbus/__init__.py

```diff
@@ -194,3 +194,24 @@ async def async_remove_entry(hass: HomeAssistant, entry: ConfigEntry) -> None:
         shutil.rmtree,
         hass.config.path(STORAGE_DIR, f"velbuscache-{entry.entry_id}"),
     )
+
+
+async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Migrate old entry."""
+    _LOGGER.debug("Migrating from version %s", config_entry.version)
+
+    if config_entry.version == 1:
+        # This is the config entry migration for adding the new program selection
+        # clean the velbusCache
+        if os.path.isdir(f"velbuscache-{config_entry.entry_id}"):
+            await hass.async_add_executor_job(
+                shutil.rmtree,
+                hass.config.path(STORAGE_DIR, f"velbuscache-{config_entry.entry_id}/"),
```

Comment:
You sure there is no storage utility function to do this?

Reply:
That's the migration that we are planning on doing. 
Currently the velbusaio module does not use the Hass storage interface but has its own pickle interface.

Reply:
I see.  Looking forward to seeing it happen.